src/internal/operators/throttleTime.ts

```diff
@@ -47,15 +47,15 @@ import { timer } from '../observable/timer';
  * internally by the optional `scheduler`.
  * @param scheduler The {@link SchedulerLike} to use for
  * managing the timers that handle the throttling. Defaults to {@link asyncScheduler}.
- * @param config a configuration object to define `leading` and
+ * @param config A configuration object to define `leading` and
  * `trailing` behavior. Defaults to `{ leading: true, trailing: false }`.
  * @return A function that returns an Observable that performs the throttle
  * operation to limit the rate of emissions from the source.
  */
 export function throttleTime<T>(
   duration: number,
   scheduler: SchedulerLike = asyncScheduler,
```

Comment:
Should `throttleTime` merge `scheduler` and `config` parameters effectively introducing a `ThrottleTimeConfig` interface which would extend `ThrottleConfig` interface by adding `scheduler?: SchedulerLike` property to it?

It would effectively remove the need to use `throttleTime` like this: `throttleTime(1000, undefined, { trailing: true })` to `throttleTime(1000, { trailing: true })`.

@benlesh, thoughts?

Reply:
Yes, this should probably be `throttleTime(number, config)` where `config` has a `scheduler` option. We'll need to deprecate the other signatures that take a scheduler as the second argument, but allow the scheduler to be passed with the config.